api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -297,6 +297,7 @@ public void purgePerson(Person person) throws APIException {
 	public Person savePerson(Person person) throws APIException {
 		setPreferredPersonName(person);
 		setPreferredPersonAddress(person);
+		Context.updateSearchIndexForType(PersonAttribute.class);
```

Comment:
are u here intending to update the index!! but i see your fields are on Person.class and no where on this class